examples/mnist/train.cpp

```diff
@@ -79,20 +87,20 @@ void train_lenet(std::string data_dir_path) {
 
     std::cout << "start training" << std::endl;
 
-    progress_display disp(train_images.size());
+    progress_display disp((unsigned long)train_images.size());
     timer t;
     int minibatch_size = 10;
     int num_epochs = 30;
 
-    optimizer.alpha *= std::sqrt(minibatch_size);
+    optimizer.alpha *= tiny_cnn::float_t(std::sqrt(minibatch_size));
 
     // create callback
     auto on_enumerate_epoch = [&](){
         std::cout << t.elapsed() << "s elapsed." << std::endl;
         tiny_cnn::result res = nn.test(test_images, test_labels);
         std::cout << res.num_success << "/" << res.num_total << std::endl;
 
-        disp.restart(train_images.size());
+        disp.restart((unsigned long)train_images.size());
         t.restart();
```

Comment:
I've found intersting comment in this file.
http://www.boost.org/doc/libs/1_61_0/boost/progress.hpp

> // NOTE: (Jan 12, 2001) Tried to change unsigned long to boost::uintmax_t, but
> // found some compilers couldn't handle the required conversion to double.
> // Reverted to unsigned long until the compilers catch up.